core/src/main/java/eu/fasten/core/plugins/FastenPlugin.java

```diff
@@ -34,57 +34,62 @@
  *     <li>The server invokes functionality on concrete plug-ins.</li>
  *     <li>The {@link :stop() method is invoked when the server is about to shutdown}</li>
  * </ol>
- *
  */
 public interface FastenPlugin extends ExtensionPoint {
 
     /**
-     * Returns a unique name for the plug-in
+     * Returns a unique name for the plug-in.
      *
      * @return The plugin's fully qualified name
      */
-    public String name();
+    String name();
 
     /**
-     * Returns a longer description of the plug-in functionality
+     * Returns a longer description of the plug-in functionality.
      *
-     * @return A string describing what the plug-in does.
+     * @return A string describing what the plug-in does
      */
-    public String description();
+    String description();
+
+    /**
+     * Returns a version of the plug-in.
+     *
+     * @return A string containing a version of the plug-in
+     */
+    String version();
 
     /**
      * Lifecycle operations: Invoked when the server has allocated all resources required to run
-     * this plug-in
+     * this plug-in.
      */
-    public void start();
+    void start();
 
     /**
      * Invoked when the server is about to shutdown. This enables the plug-in to shutdown gracefully.
      */
-    public void stop();
+    void stop();
 
     /**
-     * These two methods should be implemented so that the FASTEN server can retrieve the plug-in's error and its exception
-     * type. This will help to reprocess certain types of failed records.
+     * These two methods should be implemented so that the FASTEN server can retrieve
+     * the plug-in's error and its exception type. This will help to reprocess certain types of
+     * failed records.
+     *
+     * @param throwable exception or error encountered during plugin execution
      */
-    public void setPluginError(Throwable throwable);
+    void setPluginError(Throwable throwable);
```

Comment:
I don't understand what the difference between `set` and `get` `PluginError` is

Reply:
I didn't get this comment. `set` method is used with each plugin to save exception that occurred in a variable. Then `get` method is used to retrieve this exception. Maybe it will be a good ides to make `get` method return Optional<Throwable>.

Reply:
Why do we need the `set` method? From an outsider perspective isn't `get` enough?

Reply:
Technically we don't need it outside plugins, but I assume it was put there (which happened before I did any changes to the server) because every plugin uses this method to update its error field. I can delete it from the interface if needed.

Reply:
Sure, but in interfaces we only need to have methods that outside clients need and just them. Clients should not care about how plug-ins do set their error messages. I 'd remove `setPluginError`

Reply:
Removed